pkg/ratelimit/config.go

```diff
@@ -19,49 +19,51 @@ import (
 
 	"github.com/throttled/throttled"
 	"github.com/throttled/throttled/v2/store/memstore"
+	"go.thethings.network/lorawan-stack/v3/pkg/config"
 	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+	"gopkg.in/yaml.v2"
 )
 
 // defaultMaxSize for the rate limiter store.
 const defaultMaxSize = 1 << 12
 
-// Profile represents configuration for a rate limiting class.
-type Profile struct {
-	Name         string   `name:"name" description:"Rate limiting class name"`
-	MaxPerMin    uint     `name:"max-per-min" description:"Maximum allowed rate (per minute)"`
-	MaxBurst     uint     `name:"max-burst" description:"Maximum rate allowed for short bursts"`
-	Associations []string `name:"associations" description:"List of classes to apply this profile on"`
-}
-
-// MemoryConfig represents configuration for the in-memory rate limiting store.
-type MemoryConfig struct {
-	MaxSize uint `name:"max-size" description:"Maximum store size for the rate limiter"`
-}
-
-// Config represents configuration for rate limiting.
-type Config struct {
-	Memory   MemoryConfig `name:"memory" description:"In-memory rate limiting store configuration"`
-	Profiles []Profile    `name:"profiles" description:"Rate limiting profiles"`
-}
-
 var errRateLimitExceeded = errors.DefineResourceExhausted("rate_limit_exceeded", "rate limit of `{rate}` accesses per minute exceeded for resource `{key}`")
 
 // New creates a new ratelimit.Interface from configuration.
-func (c Config) New(ctx context.Context) (Interface, error) {
+func New(ctx context.Context, conf config.RateLimiting, blobConf config.BlobConfig) (Interface, error) {
 	defaultLimiter := &NoopRateLimiter{}
-	if len(c.Profiles) == 0 {
+	profiles := conf.Profiles
+
+	fetcher, err := conf.Fetcher(ctx, blobConf)
+	if err != nil {
+		return nil, err
+	}
+	if fetcher != nil {
+		b, err := fetcher.File("rate-limiting.yml")
+		if err != nil {
+			return nil, err
+		}
+		var overrideProfiles struct {
+			Profiles []config.RateLimitingProfile `yaml:"profiles"`
+		}
+		if err := yaml.Unmarshal(b, &overrideProfiles); err != nil {
+			return nil, err
+		}
+		profiles = append(profiles, overrideProfiles.Profiles...)
+	}
+	if len(profiles) == 0 {
 		return defaultLimiter, nil
 	}
 
 	l := &muxRateLimiter{
 		defaultLimiter: defaultLimiter,
-		limiters:       make(map[string]Interface, len(c.Profiles)),
+		limiters:       make(map[string]Interface, len(profiles)),
 	}
-	for _, profile := range c.Profiles {
+	for _, profile := range profiles {
```

Comment:
So you merge the profiles here from the various files?

Reply:
Yes. Profiles from external configuration are set last, and so will take priority.